src/Microsoft.AspNet.Server.Kestrel/Http/Listener.cs

```diff
@@ -1,25 +1,22 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using Microsoft.AspNet.Server.Kestrel.Infrastructure;
 using Microsoft.AspNet.Server.Kestrel.Networking;
 using System;
 using System.Diagnostics;
-using System.Net;
 using System.Threading.Tasks;
 
 namespace Microsoft.AspNet.Server.Kestrel.Http
 {
     /// <summary>
-    /// Summary description for Accept
+    /// Base class for listeners in Kestrel. Listens for incoming connections
     /// </summary>
-    public class Listener : ListenerContext, IDisposable
+    /// <typeparam name="T">Type of socket used by this listener</typeparam>
+    public abstract class Listener<T> : ListenerContext, IListener where T : UvStreamHandle
     {
-        private static readonly Action<UvStreamHandle, int, Exception, object> _connectionCallback = ConnectionCallback;
+        protected T ListenSocket { get; private set; }
```

Comment:
It was just in case I needed to call a method specific to the particular implementation. I don't think I ended up needing to do that though, so it could probably just be `UvStreamHandle`.